quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java

```diff
@@ -156,18 +157,26 @@ public <T> T set(T value) {
                 addHelp(currentSpec);
             }
 
-            AbstractCommand currentCommand = null;
+            AbstractCommand currentCommand = null; // this is the command that will be used in the next parsing pass
+            CommandLine commandLine = null;
             if (currentSpec != null) {
-                CommandLine commandLine = currentSpec.commandLine();
-                addCommandOptions(cliArgs, commandLine);
+                commandLine = currentSpec.commandLine();
 
                 if (commandLine != null && commandLine.getCommand() instanceof AbstractCommand ac) {
                     currentCommand = ac;
                 }
             }
-            initConfig(currentCommand);
+            // init the config before adding options to properly sanitize mappers
+            boolean allHelpOptions = false;
+            if (commandLineList.size() > 1 && commandLineList.get(commandLineList.size() - 1).getCommand() instanceof AbstractCommand ac) {
+                allHelpOptions = Optional.ofNullable(ac.getHelpAllMixin()).map(HelpAllMixin::isAllOptions).orElse(false);
+            }
+            initConfig(allHelpOptions, currentCommand);
```

Comment:
Cannot we get the info about the allHelp from the `currentCommand`? Does it have to be an additional argument of the `initConfig()`?